src/modules/auth/credentials/BulkAssignmentContainer.js

```diff
@@ -0,0 +1,485 @@
+					tenantsTableData.push(matchObj);
+				} else {
+					tenantObj['assigned'] = false;
+					tenantsTableData.push(tenantObj);
+				}
+			})
+		}
+		return tenantsTableData
+	}, [tenants, selectedTenants]);
+
+	// fetches Credentials Data from server
+	const retrieveData = async () => {
+		setLoading(true);
+		try {
+			let SeaCatAuthAPI = props.app.axiosCreate('seacat_auth');
+			let response = await SeaCatAuthAPI.get("/credentials", {params: {p:page, i: limit, f: credentialsFilter}});
+			if (response.data.result !== "OK") {
+				throw new Error(t("BulkAssignmentContainer|Failed to fetch credentials"));
+			};
+			setData(response.data.data);
+			setCount(response.data.count);
+			setLoading(false);
+		} catch(e) {
+			console.error(e);
+			setLoading(false);
+			// error message for unauthorized access
+			if (e.response.status === 401) {
+				props.app.addAlert("warning", t("BulkAssignmentContainer|Can't fetch credentials, you don't have rights to display it"), 30);
+				return;
+			};
+			props.app.addAlert("warning", `${t("BulkAssignmentContainer|Failed to fetch credentials")}. ${e?.response?.data?.message}`, 30);
+		};
+	};
+
+	// fetch data for Tenants List from server
+	const retrieveTenants = async () => {
+		setLoadingTenants(true);
+		try {
+			let SeaCatAuthAPI = props.app.axiosCreate('seacat_auth');
+			let response = await SeaCatAuthAPI.get("/tenants", {params: {p: tenantsPage, i: tenantsLimit, f: tenantsFilter}});
+			setTenants(response.data.data);
+			setTenantsCount(response.data.count);
+		} catch(e) {
+			console.error(e);
+			props.app.addAlert("warning", `${t("BulkAssignmentContainer|Failed to fetch tenants")}. ${e?.response?.data?.message}`, 30);
+		}
+		setLoadingTenants(false);
+	};
+
+	// call to assign all selected tenants to all selected credentials
+	const bulkAction = async (actionType) => {
+		let credential_ids = [];
+		let tenantObj = {};
+		// adjustments to data structure. selectedCredentials is an array of objects. Server expects credential_ids to be an array of ids only [ 'id1', 'id2', ..]
+		selectedCredentials.map((obj) => {
+			credential_ids.push(obj._id);
+		})
+		/* adjustments to data structure. selectedTenants is an array of objects. Server expects credential_ids to be an object with
+		tenants as keys and array of roles as their values { tenant1: [role1, role2], tenant2: [], ...}
+		Unless we'd like to do bulk *unassignment*, then unassigning from tenant as a whole requires { tenant1: "UNASSIGN-TENANT", tenant2: [role1, role2,..], ...} */
+		// let globalRoles = [];
+		selectedTenants.map((obj) => {
+			console.log('obj: ', obj);
+			let roles = [];
+			if (obj.selectedRole && (obj.selectedRole?.length > 0)) {
+					roles = obj.selectedRole;
+			} else if (actionType === '/tenant_unassign_many') {
+				roles = "UNASSIGN-TENANT";
+			};
+			// here in code, we are using `{_id: Global roles, ...}` which represents global roles, but the api call expects `_id: '*'` as a representation of global roles
```

Comment:
@petrKavulok Try to write comments more object based - meaning - its always better not to use any personification, when writing comments about the code - so, not `we`, `I`, ... but rather writ about what it does and why it is used. eg.

`// This part of the code uses `{_id: Global roles, ...}`, which is the representation of global roles, however the API call expects `_id` to be defined as asterisk `'*'` as a representation of global roles`